### DISCUSSION TITLE
Wine apps on Meson

### DISCUSSION THREAD

**Discussion by mlauronen** [mesonbuild/meson#12457] [Q&A] [answered]
I'm trying to learn how to configure and cross compile a simple c++ hello world level application on Meson when the build machine is Debian Linux and the host machine is Linux Wine.

The following command does the thing from CLI but how to configure and compile the thing on Meson ?
`
86_64-w64-mingw32-c++ -D _WINDOWS -D _USRDLL -D _WINDLL -w -m64 -march=native -std=c++17 -static -I ./headers hello.cpp -o hello.exe
`
The -I parameter is used here because in real project I will need it.

Could you please give me some advice and guidance on how to start dealing with this ?
Maybe an example project somewhere ?

**Comment by xclaesse** [comment]
`meson setup builddir --cross-file windows.ini`, you can find an example of cross file for example: https://gitlab.freedesktop.org/gstreamer/gstreamer/-/blob/main/data/machine-files/mingw_w64_x86-64-cross-file.ini?ref_type=heads

**Reply by mlauronen** [reply]
Don't I have to create meson.build first where I will tell what we have to compile and the parameters ?

That 'meson setup...' command generates the following message:
`ERROR: Neither source directory 'builddir' nor build directory None contain a build file meson.build.`

**Reply by xclaesse** [reply]
Yes, of course, but for that I would need to know what project you're working on...

**Reply by mlauronen** [reply]
I thought the CLI command will tell you.... just tell me what do you want to know more ?
Application is a simple Hello World! writer. Just trying to understand the configuration logic here.

**Reply by xclaesse** [reply, marked as answer]
You can use `meson init` to help generating a first template for your project. The read the doc, simple examples: https://mesonbuild.com/Meson-sample.html.

**Reply by mlauronen** [reply]
Ok. I think I got the idea now. The compile instructions will be declared in meson.build and the cross-compile settings in --cross-file text file. 

Thank you.
